src/khoj/configure.py

```diff
@@ -99,12 +125,24 @@ def configure_search(model: SearchModels, config: FullConfig, regenerate: bool,
         )
 
     # Initialize Image Search
-    if (t == SearchType.Image or t == None) and config.content_type.image:
+    if (t == state.SearchType.Image or t == None) and config.content_type.image:
         # Extract Entries, Generate Image Embeddings
         model.image_search = image_search.setup(
             config.content_type.image, search_config=config.search_type.image, regenerate=regenerate
         )
 
+    # Initialize External Plugin Search
```

Comment:
If I've understood this correctly, the user would still have to set up an independent system on their side for periodically regenerating the jsonl file, correct?

So the plugin basically has to handle the portion of processing the target text file + dumping it in the appropriate jsonl format, and then it'll get picked up by `Khoj`'s search configuration.

It might be neat to include a sample skeleton of a basic plugin for users to get started. Non-blocking.